services/src/datasets/add_from_directory.rs

```diff
@@ -74,29 +79,48 @@ pub async fn add_providers_from_directory<D: LayerProviderDb>(db: &mut D, file_p
         Ok(())
     }
 
-    let dir = fs::read_dir(file_path);
-    if dir.is_err() {
-        warn!("Skipped adding providers from directory because it can't be read");
-        return;
-    }
-    let dir = dir.expect("checked");
+    async fn add_from_dir<D: LayerProviderDb>(db: &mut D, file_path: &Path) {
+        let dir = fs::read_dir(file_path);
+        if dir.is_err() {
+            warn!(
+                "Skipped adding providers from directory `{:?}` because it can't be read",
+                file_path
+            );
+            return;
+        }
+        let dir = dir.expect("checked");
 
-    for entry in dir {
-        if let Ok(entry) = entry {
-            if entry.path().is_dir() {
-                continue;
-            }
-            if let Err(e) = add_provider_definition_from_dir_entry(db, &entry).await {
-                // TODO: log
-                warn!(
-                    "Skipped adding provider from directory entry: {:?} error: {}",
-                    entry,
-                    e.to_string()
-                );
+        for entry in dir {
+            match entry {
+                Ok(entry) if entry.path().is_file() => {
+                    match add_provider_definition_from_dir_entry(db, &entry).await {
+                        Ok(_) => info!("Added provider from file `{:?}`", entry.path()),
+                        Err(e) => {
+                            warn!(
+                                "Skipped adding provider from file `{:?}` error: `{}`",
+                                entry.path(),
+                                e.to_string()
+                            );
+                        }
+                    }
+                }
+                Err(e) => {
+                    // TODO: log
+                    warn!("Skipped adding provider from directory entry `{:?}`", e);
+                }
+                _ => {}
             }
-        } else {
-            // TODO: log
-            warn!("Skipped adding provider from directory entry: {:?}", entry);
         }
     }
+
+    add_from_dir(db, &file_path).await;
+
+    #[cfg(feature = "pro")]
+    add_from_dir(db, &file_path.join("pro")).await;
```

Comment:
We said our goal was to have only one entry-point for pro.
Can we somehow keep it this way for this place?

Reply:
we could duplicte the whole method in pro and only add the "pro" subfolder there.

or accept multiple `file_path`s and put only the base folder in, for non-pro, and the pro-folder additionaly in pro

Reply:
The last thing sounds better than duplicating everything!

Reply:
b0a9416f44a392c08b060cd1a4fe013523efc9b6